platform-sdk/swirlds-common/src/main/java/com/swirlds/common/formatting/UnitFormatter.java

```diff
@@ -160,87 +158,177 @@ public UnitFormatter setShowSpaceInBetween(final boolean showSpaceInBetween) {
      *
      * @return the rendered string
      */
+    @NonNull
     public String render() {
         final StringBuilder sb = new StringBuilder();
         render(sb);
         return sb.toString();
     }
 
     /**
-     * Render the number and write it to a string builder
+     * Checks whether the given quantity string requires a plural unit
+     * <p>
+     * Any number other than 1.0 is considered to be plural.
      *
-     * @param sb
-     * 		the string builder to write the rendered string to
+     * @param quantityString the quantity string
+     * @return true if the quantity string requires a plural unit
      */
-    public void render(final StringBuilder sb) {
+    private boolean isQuantityPlural(@NonNull final String quantityString) {
+        boolean plural = false;
 
-        if (longQuantity == null && doubleQuantity == null) {
-            throw new IllegalStateException("Quantity must be set");
+        if (quantityString.charAt(0) != '1') {
+            plural = true;
+        } else if (quantityString.length() > 1) {
+            if (quantityString.charAt(1) != '.') {
+                plural = true;
+            } else {
+                for (int i = 2; i < quantityString.length(); i++) {
+                    final char c = quantityString.charAt(i);
+                    if (c != '0') {
+                        plural = true;
+                        break;
+                    }
+                }
+            }
```

Comment:
```suggestion
        return Double.parseDouble(quantityString) != 1;
```
(Note: this may not work if the quantity string contains `,` characters)

Reply:
Also, it seems kind of like a code smell that we are parsing back strings we generated from numeric types. Could we instead just use the original numbers before they are converted to strings? It looks like this is code that was preexisting that you just rearranged... so if this requires major refactor perhaps we don't do it now. (Side note: I originally wrote this class on one of my weekend hackathon sessions, so it may not be the cleanest in some places)

Reply:
Fixed